source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -73,6 +73,7 @@ namespace Type
     const TypeDesc* BSDF               = TypeDesc::registerType("BSDF", TypeDesc::BASETYPE_NONE, TypeDesc::SEMATIC_CLOSURE, 1, false);
     const TypeDesc* EDF                = TypeDesc::registerType("EDF", TypeDesc::BASETYPE_NONE, TypeDesc::SEMATIC_CLOSURE, 1, false);
     const TypeDesc* VDF                = TypeDesc::registerType("VDF", TypeDesc::BASETYPE_NONE, TypeDesc::SEMATIC_CLOSURE, 1, false);
+    const TypeDesc* ROUGHNESSINFO      = TypeDesc::registerType("roughnessinfo", TypeDesc::BASETYPE_NONE, TypeDesc::SEMATIC_NONE, 1, false);
```

Comment:
I'm not set on the naming of the new type yet. I didn't want to use just "roughness" since it so commonly used as a variable name. Other names I considered are: roughnessdesc, surfaceroughness, mfroughness (mf for microfacet), but these felt too long or just not as clear as roughnessinfo.